src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactory.cs

```diff
@@ -471,6 +461,20 @@ private static Type GetReturnMarshaler(MethodInfo method, out Func<object?, Valu
 #endif
         }
 
+        private static readonly MethodInfo _taskGetResult = typeof(Task<>).GetProperty(nameof(Task<int>.Result), BindingFlags.Instance | BindingFlags.Public)!.GetMethod!;
+        private static readonly MethodInfo _valueTaskAsTask = typeof(ValueTask<>).GetMethod(nameof(ValueTask<int>.AsTask), BindingFlags.Instance | BindingFlags.Public)!;
+
+        [UnconditionalSuppressMessage("Trimming", "IL2070:'this' argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method.",
+            Justification = "The MethodInfo we are looking for must have already been rooted by virtue of its generic definition being available.")]
+        private static MethodInfo GetMethodFromGenericMethodDefinition(Type specializedType, MethodInfo genericMethodDefinition)
```

Comment:
@eerhardt NB I implemented this from scratch because we're also targeting netstandard2.0

Reply:
Writing a bespoke helper seemed faster than extracting a polyfill here given how trivial it is.